packages/material-ui/src/Select/SelectInput.js

```diff
@@ -67,48 +67,54 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
   });
 
   const inputRef = React.useRef(null);
+  const displayRef = React.useRef(null);
   const [displayNode, setDisplayNode] = React.useState(null);
   const { current: isOpenControlled } = React.useRef(openProp != null);
   const [menuMinWidthState, setMenuMinWidthState] = React.useState();
   const [openState, setOpenState] = React.useState(false);
   const handleRef = useForkRef(ref, inputRefProp);
 
+  const handleDisplayRef = React.useCallback((node) => {
+    displayRef.current = node;
+
+    if (node) {
+      setDisplayNode(node);
```

Comment:
I want to investigate this first. I'm not sure why this is continually marked as an easy issue when it's a really complicated solution. A state update during ref phase is a last resort which #21443 didn't need to do. We should make sure we're not opting for a quick solution that makes the 90% work but the 10% infinitely harder to solve.

Reply:
This idea behind the change is to use as much as possible the nominal case: storing refs in a useRefs, while using state only when the value is used during rendering. I don't understand the case you are considering.

Reply:
@eps1lon What do you think?

Reply:
It's wrong but you don't care so let's ship the wrong/slow behavior.

Reply:
I think that we can wait for feedback from developers about it. If they report regressions, we can investigate or revert.

Reply:
I understand you wanted to look deeper into it, I also understand that it wasn't a priority, I think that it's fine, we can defer the exploration to developers, they will stress-test this :)